raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -1073,37 +1082,35 @@ def handle_refundtransfer(
     iteration = TransitionResult(mediator_state, list())
 
     if mediator_state.secret is None:
-        # The last sent transfer is the only one thay may be refunded, all the
+        # The last sent transfer is the only one that may be refunded, all the
         # previous ones are refunded already.
-        transfer_pair = mediator_state.transfers_pair[-1]
-        payee_transfer = transfer_pair.payee_transfer
+        payer_transfer = mediator_state_change.transfer
+        channel_address = payer_transfer.balance_proof.channel_address
+        payer_channel = channelidentifiers_to_channels[channel_address]
+        is_valid, events, _ = channel.handle_refundtransfer(
+            payer_channel,
+            mediator_state_change,
+        )
+        if not is_valid:
+            return TransitionResult(None, events)
 
-        is_valid = is_valid_refund(
-            payee_transfer,
-            mediator_state_change.transfer,
+        iteration = mediate_transfer(
+            mediator_state,
+            mediator_state_change.routes,
+            payer_channel,
+            channelidentifiers_to_channels,
+            pseudo_random_generator,
+            payer_transfer,
+            block_number,
         )
-        if is_valid:
-            payer_transfer = mediator_state_change.transfer
-            channel_address = payer_transfer.balance_proof.channel_address
-            payer_channel = channelidentifiers_to_channels[channel_address]
-
-            iteration = mediate_transfer(
-                mediator_state,
-                mediator_state_change.routes,
-                payer_channel,
-                channelidentifiers_to_channels,
-                pseudo_random_generator,
-                payer_transfer,
-                block_number,
-            )
 
-            events = list(iteration.events)
-            send_processed = SendProcessed(
-                mediator_state_change.transfer.balance_proof.sender,
-                b'global',
-                mediator_state_change.message_identifier,
-            )
-            events.append(send_processed)
+        events = list(iteration.events)
+        send_processed = SendProcessed(
```

Comment:
This is being done inside the `channel.handle_refundtransfer`, so it's not required here anymore